pkg/node/client.go

```diff
@@ -0,0 +1,165 @@
+package node
+
+import (
+	"context"
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog"
+
+	"github.com/kubernetes-sigs/container-object-storage-interface-api/apis/objectstorage.k8s.io/v1alpha1"
+	cs "github.com/kubernetes-sigs/container-object-storage-interface-api/clientset/typed/objectstorage.k8s.io/v1alpha1"
+)
+
+type NodeClient struct {
+	cosiClient *cs.ObjectstorageV1alpha1Client
+	kubeClient kubernetes.Interface
+}
+
+func NewClient() (*NodeClient, error) {
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	client := cs.NewForConfigOrDie(config)
+	kube := kubernetes.NewForConfigOrDie(config)
+	return &NodeClient{
+		cosiClient: client,
+		kubeClient: kube,
+	}, nil
+}
+
+const (
+	barNameKey      = "bar-name"
+	barNamespaceKey = "bar-namespace"
+)
+
+func parseValue(key string, ctx map[string]string) (string, error) {
+	value, ok := ctx[key]
+	if !ok {
+		return "", fmt.Errorf("required volume context key unset: %v", key)
+	}
+	klog.Infof("got value: %v", value)
+	return value, nil
+}
+
+func parseVolumeContext(volCtx map[string]string) (name, ns string, err error) {
+	klog.Info("parsing bucketAccessRequest namespace/name from volume context")
+	name, err = parseValue(barNameKey, volCtx)
+	if err != nil {
```

Comment:
nit: slightly cleaner/more compact as:
```
if name, err = parseValue(barNameKey, volCtx); err != nil { ...
```

Reply:
and as a personal preference, even when I name the return args, I like to name them explicitly in the `return`, so
```
   if name, err = parseValue(barNameKey, volCtx); err != nil { 
        return "", "", err
   }
```

Reply:
Should I remove the named return args in this case and just use the explicit return values?

Reply:
Named return args are most useful in multiple-assignments, at least for me. Eg something like;
```
func xyz() (arg1 string, err error) {
     var foo int
     foo, arg1, err := a_func()
}
```

 I think the code is more readable when the `return`s include the return args inline.

Reply:
Makes sense! Fixed